sqlalchemy_hana/dialect.py

```diff
@@ -69,16 +69,19 @@ def limit_clause(self, select, **kwargs):
         return text
 
     def for_update_clause(self, select, **kwargs):
-        tmp = " FOR UPDATE"
+        if select._for_update_arg.read:
+            tmp = " FOR SHARE LOCK"
```

Comment:
In the current implementation, the dialect would silently ignore the other parameters like `NOWAIT` or ` OF ...`. What do you think, should the dialect prevent wrong usage and raise an exception that is visible the applications test suite or while development?

Reply:
Valid point. If the other dialects silently ignore the error then we should follow. Maybe add a small comment about that into the source code. :)

Reply:
I added some comments here. what do you think?

Reply:
Looks good to me.